algo/src/main/java/org/neo4j/graphalgo/UnionFindProc.java

```diff
@@ -0,0 +1,108 @@
+package org.neo4j.graphalgo;
+
+import algo.Pools;
+import com.carrotsearch.hppc.IntIntMap;
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.core.GraphLoader;
+import org.neo4j.graphalgo.core.heavyweight.HeavyGraphFactory;
+import org.neo4j.graphalgo.core.utils.dss.DisjointSetStruct;
+import org.neo4j.graphalgo.impl.GraphUnionFind;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+
+import java.util.Map;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author mknblch
+ */
+public class UnionFindProc {
+
+    public static final String CONFIG_PROPERTY = "property";
+    public static final String CONFIG_DEFAULT_VALUE = "defaultValue";
+    public static final String CONFIG_THRESHOLD = "threshold";
+
+    @Context
+    public GraphDatabaseAPI api;
+
+    @Context
+    public Log log;
+
+    @Procedure(value = "algo.unionFind")
+    @Description("CALL algo.unionFind(label:String, relationship:String, {property:'propertyName', threshold:0.42, defaultValue:1.0}) " +
+            "YIELD nodeId, setId - yields a setId to each node id")
+    public Stream<DisjointSetStruct.Result> unionFind(
+            @Name("label") String label,
```

Comment:
label and rel-type should still have a defaultValue=""
if you want to run it on the whole graph